modules/button/react/lib/types.ts

```diff
@@ -92,3 +92,21 @@ export interface ButtonColors {
   };
   disabled: ButtonStateColors;
 }
+
+/**
+ * Used to get the props of the anchor version of a button
+ */
+export type AnchorButtonProps<P> = Omit<P, keyof React.ButtonHTMLAttributes<HTMLButtonElement>> &
+  React.AnchorHTMLAttributes<HTMLAnchorElement>;
+
+/**
+ * Returns an overloaded functional component that uses props P by default,
+ * but uses AnchorButtonProps<P> when `as="a"`.
+ * Note: V must be `typeof ButtonVariant` since TS cannot assign enums as a type in a generic properly
+ */
+export type ButtonOrAnchorComponent<P, V = undefined> = {
+  (props: P): React.ReactElement;
+  (props: {as: 'a'} & AnchorButtonProps<P>): React.ReactElement;
```

Comment:
Not sure if this is a restriction with the way TS overloads work, but anchor attributes aren't allowed when `as` is not defined, but when it is defined, button attributes are still allowed.

Reply:
That's interesting. I don't get that behavior: [Example](https://www.typescriptlang.org/play/?jsx=2#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wChSkA7AVzwCFqYYJKA1FKYFS+Ab1LhwACpxAcAnnAC8BMKIlkAvuSq04AERxgAJhADulBkxbtO3PgLgAZDgHMk0ggBs7SJeVLAeSKJnQOjZkoRCDAAZzgkAA8YKm0I5HQYADpAlgAJABUAWSsAQSZOACNGJDCAHizctMoAUSckECoYAD44fkEANw4uHgAuOBrTXphSZU9vX38NLV0DGpDwyJi4iIB5EGAYcoWtMIAaAm6zHnw2jrhjkYHNULnDRiDh8zHyGHEwBzzKNAALaF2oQqdDaMg2Wx2hwA1khxBBMIhiCkalV8oVgCVYhVUTV6o1mi02gAyRFJZLfP7QVEFGDFUrYnL5H7-KB4po8FoqKKQWBwd6fQaPFhrTo+JwQFDaSFwVig9qWAAUciBAzoAEoBokMMktTA2c1FcrwgNeHAUGEBvgUPg4Io4CSKSzAeEdi0NaTtbr9TxLM9+jLXuQ0CwwvBbjp9A9jJRVULKCKxRKpeH7s6DnyPkh4TM7pGhj1zHKFWrpG1ytpgJ04AB6TkpvNx5J++AyevzOPN8gNeCxcCkXtgRzlNtRoJwTDQEBSABEE4g08uBZ4Ul4I5qTaXKRsUHsihaADEIBA6BxytW13HOQOhxfo+PJzO5wvzTOUAuruYV7eghuTlvXHuh7Hqe56zA20acv2jSDjIw5ge2d6-EQmCPke76bl+8Gjiwv4jMk267geR4nlAZ7fiwV7QTeWE1HASFIChs5oWaYSvuhf6YbmCE-s2+EAURwGkaBXHYZQLRAA). Perhaps the TS version is causing this difference?